packages/clerk-js/src/utils/queryStateParams.ts

```diff
@@ -0,0 +1,40 @@
+import { getClerkQueryParam } from '../utils';
+
+export const decodeBase64Json = (value: string) => {
+  return JSON.parse(atob(value));
+};
+
+export const encodeBase64Json = (value: { path: string; componentName: string }) => {
+  return btoa(JSON.stringify(value));
+};
+
+export const readAndRemoveStateParam = () => {
```

Comment:
❓ Does this function also remove the __clerk_state param from the URL or just reads it?